# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/find/test/mochitest/test_nsFind.html
# Commit: e07e5ea2f75e
# Full Hash: e07e5ea2f75e971967decbbe8f42aff80987fda2
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-02-16 08:52:20
# Regressor Bug: 1945862
# File Overlap Count: 1
# Description:
#   Bug 1945862 - Make `nsFind` respect the end point correctly. r=emilio
#   
#   This patch refactors the find functionality in a few ways:
#   - A C++-only method `FindFromRangeBoundaries()` is introduced,
#     which uses `RangeBoundary`s instead of collapsed `nsRange`s
# ==============================================================================

diff -r 5a1f8c0bb78f -r e07e5ea2f75e toolkit/components/find/test/mochitest/test_nsFind.html
--- a/toolkit/components/find/test/mochitest/test_nsFind.html	Sat Feb 15 15:19:21 2025 +0000
+++ b/toolkit/components/find/test/mochitest/test_nsFind.html	Sat Feb 15 16:02:34 2025 +0000
@@ -40,9 +40,10 @@
                         .getService(SpecialPowers.Ci.nsIFind);
 
   var display = window.document.getElementById("display");
+  var testNode = window.document.getElementById("test");
   var searchRange = window.document.createRange();
   searchRange.setStart(display, 0);
-  searchRange.setEnd(display, display.childNodes.length);
+  searchRange.setEnd(testNode, testNode.childNodes.length);
   var startPt = searchRange;
   var endPt = searchRange;
 
@@ -145,23 +146,26 @@
   ok(retRange, "\"" + searchValue + "\" not found");
 
   // Set new start point (to end of last search).
-  startPt = retRange.endContainer.ownerDocument.createRange();
-  startPt.setStart(retRange.endContainer, retRange.endOffset);
-  startPt.setEnd(retRange.endContainer, retRange.endOffset);
+  let newStartPt = retRange.endContainer.ownerDocument.createRange();
+  newStartPt.setStart(retRange.endContainer, retRange.endOffset);
+  newStartPt.setEnd(retRange.endContainer, retRange.endOffset);
 
   searchValue = "t";
-  retRange = rf.Find(searchValue, searchRange, startPt, endPt);
+  retRange = rf.Find(searchValue, searchRange, newStartPt, endPt);
   ok(retRange, "\"" + searchValue + "\" not found (forward)");
 
   searchValue = "the";
-  retRange = rf.Find(searchValue, searchRange, startPt, endPt);
+  retRange = rf.Find(searchValue, searchRange, newStartPt, endPt);
   ok(!retRange, "\"" + searchValue + "\" found (forward)");
 
   rf.findBackwards = true;
 
   // searchValue = "the";
-  retRange = rf.Find(searchValue, searchRange, startPt, endPt);
-  ok(retRange, "\"" + searchValue + "\" not found (backward)");
+  retRange = rf.Find(searchValue, searchRange, newStartPt, endPt);
+  ok(!retRange, "\"" + searchValue + "\" found in second sub range (backward)");
+
+  retRange = rf.Find(searchValue, searchRange, startPt, newStartPt);
+  ok(retRange, "\"" + searchValue + "\" not found in first sub range (backward)");
 
   // Curly quotes and straight quotes should match.
 